lib/utils.js

```diff
@@ -94,13 +92,34 @@ function parseParams(str) {
   return res;
 }
 
+const DecoderMap = new Map();
+
 function decodeText(text, textEncoding, destEncoding) {
-  var ret;
-  if (text && jsencoding.encodingExists(destEncoding)) {
-    try {
-      ret = jsencoding.TextDecoder(destEncoding)
-                      .decode(Buffer.from(text, textEncoding));
-    } catch(e) {}
+  let ret;
+  /**
+   * https://datatracker.ietf.org/doc/html/rfc7578#section-4.7
+   */
+  switch (destEncoding) {
```

Comment:
if we only support two encodings, do we even need map? can have two consts and switch between them.

Reply:
not necessary :)